File: Amazon_LeetCode_Python_C++/ArrayString/186_reverse_words_in_a_string_II.cpp
class Solution {
public:
    void reverseWords(vector<char>& s) {
        reverse(s.begin(), s.end());
        int r = 0, l = 0;
        while (r < s.size()) {
            while (l < s.size() && !isblank(s[l])) {
                l++;
            }
            reverse(s.begin() + r, s.begin() + l);
            l++;
            r = l;
        }
    }
};

File: Amazon_LeetCode_Python_C++/ArrayString/186_reverse_words_in_a_string_II.py
class Solution:
    def reverseWord(self, s: list, left: int, right: int) -> None:
        while left < right:
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1
    
    def reverseWords(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        r = l = 0
        s.reverse()
        while r < len(s):
            while l < len(s) and s[l] != ' ':
                l += 1
            self.reverseWord(s, r, l - 1)
            l += 1
            r = l
